coral-hive/src/main/java/com/linkedin/coral/transformers/SingleUnionFieldReferenceTransformer.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.transformers;
+
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+import com.linkedin.coral.common.functions.FunctionFieldReferenceOperator;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+
+
+/**
+ * This transformer focuses on SqlCalls that involve a FunctionFieldReferenceOperator with the following characteristics:
+ * (1) The first operand is a SqlBasicCall with a non-struct RelDataType, and the second operand is tag_0.
+ * This indicates that the first operand represents a Union data type with a single data type inside.
+ * (2) Examples of such SqlCalls include extract_union(product.value).tag_0 or (extract_union(product.value).id).tag_0.
+ * (3) The transformation for such SqlCalls is to return the first operand.
+ */
+public class SingleUnionFieldReferenceTransformer extends SqlCallTransformer {
+  private static final String TAG_0_OPERAND = "tag_0";
+
+  public SingleUnionFieldReferenceTransformer(SqlValidator sqlValidator) {
+    super(sqlValidator);
+  }
```

Comment:
`sqlValidator` is not necessary if we don't need to call `getType(SqlNode)` method in this transformer?

Reply:
im using `getRelDataType()` [here](https://github.com/linkedin/coral/pull/409/files/162e2baa7f2cc107d8602219eab43bd15b4b42f9#diff-aca6363aefb30015f0a9c5fb6bef3aafd0f3638cc897c903e125bb67bb6f6393R35) to derive if the first operand is struct or not